### PR TITLE
Feature/kc swift 4

### PR DESCRIPTION
**Pull request by KaneCheshire**

### CODE REVIEW

File: Example/Tests/ObserverSetTests.swift [hunk 1]
```diff
@@ -20,7 +20,7 @@ class ObserverSetTests: XCTestCase {
         let namedParameterObservers = ObserverSet<(name: String, count: Int)>()
         
         func testNotify() {
-            voidObservers.notify()
+            voidObservers.notify(())
```
Comment (KaneCheshire) [inline Example/Tests/ObserverSetTests.swift]: I hate this. Any way around it?

File: Example/Tests/ObserverSetTests.swift [hunk 2]
```diff
@@ -36,7 +36,7 @@ class ObserverSetTests: XCTestCase {
             observee.twoStringObservers.add(self, type(of: self).twoStringChanged)
             observee.intObservers.add(self, type(of: self).intChanged)
             observee.intAndStringObservers.add(self, type(of: self).intAndStringChanged)
-            observee.namedParameterObservers.add(self, type(of: self).namedParameterSent)
+//            observee.namedParameterObservers.add(self, type(of: self).namedParameterSent)
```
Comment (KaneCheshire) [inline Example/Tests/ObserverSetTests.swift]: I can't get this to compile :(

File: Example/Tests/ObserverSetTests.swift (hunk 1 above)
Comment (lucianomarisi) [reply Example/Tests/ObserverSetTests.swift]: I was hoping we could pass `Void` because it's a typealias but that did not work 😞

Comment (KaneCheshire) [reply Example/Tests/ObserverSetTests.swift]: It feels wrong to do it like this :(

**Comment by KaneCheshire** [comment]
[Convert TABObserverSet to Swift 4.0](https://trello.com/c/xDNUHCKw/4-convert-tabobserverset-to-swift-40)

**Comment by KaneCheshire** [comment]
[Convert TABObserverSet to Swift 4](https://trello.com/c/IAs7ExEI/3-convert-tabobserverset-to-swift-4)

**Comment by KaneCheshire** [comment]
[Convert TABObserverSet to Swift 4](https://trello.com/c/IAs7ExEI/3-convert-tabobserverset-to-swift-4)

**Comment by KaneCheshire** [comment]
With heavy heart I have removed the test that is now impossible to compile for Swift 4. We'll just have to treat it as deprecated. Will look at the merge conflicts now.

**Review by jsanderson44 — APPROVED** [review]

**Pull request merged**